Assets/Scripts/Entity/Enemy/EnemyDrop.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyDrop : MonoBehaviour
+{
+    /* Called when an enemy has died. Spawns a heart with random probability.
+    */
+    public static void DropHeart(Transform t, float healthDropProbability, float heartTypeProbability)
+    {
+        var heartType = GameObject.Find("Half-Heart");
```

Comment:
Instead of doing this, which is a bit inefficient and also slightly impractical, I suggest giving this script a publicly exposed `List` of `GameObjects`, called `heartPrefabs`, for example. Then, in your code, you can simply generate a random index and spawn the heart that's stored at that index.

Reply:
> Alternatively to Alek's suggestion of adding code into `Enemy.cs` to spawn the hearts, you could instead of use Enemy's `deathEvent` to cause the heart drop in your EnemyDrop file.

This is actually what I was waiting for. When we discussed this previously death wasn't really handled yet.

Reply:
Not a bad idea. This list could be expanded easily too since I could just generate the upper bound for the random index based on the list size.